### PR TITLE
Merging timeseries

### PR DESCRIPTION
**Pull request by jrt54**
New timeseries functions to merge together several timeseries into one sorted timeseries and return the resulting timeseries object. Also created a convenience routine to return datetime objects which may entail slicing off the "+Z" timezone character from the timestamps. 

This works only based off of timeseries uuids.

I created tests in tests/test_timeseries.py, but I'm not sure how to configure the CI workflow to run them automatically

### CODE REVIEW

**Comment by connortann** [comment]
The static analysis failure is not your fault - looks like it wasn't set up correctly to run on forks.

I've updated the GitHub actions workflow in #35 ; so try rebasing your branch on top of master and see if it passes now.

**Comment by connortann** [comment]
@andy-beer an issue is that python 3.6 does not support `datetime.fromisoformat` . It looks like we could get this working by installing this extra pip package:

https://github.com/movermeyer/backports.datetime_fromisoformat

However, it may be simpler to just drop support for python 3.6 . Two files need to be updated:
- setup.cfg , the classifiers section for PyPI
- github/workflows/ci-tests.yml , the "matrix strategy" section for running unit tests

File: resqpy/time_series.py [hunk 1]
```diff
@@ -247,6 +247,18 @@ def extend_by_days(self, days):
...
+      datetimes=[]
+        
+      for timestamp in self.timestamps:
+         datetime_timestamp=timestamp
+         if timestamp.endswith('Z'): datetime_timestamp = timestamp[:-1]
+         datetimes.append(dt.datetime.fromisoformat(datetime_timestamp))
+        
+        
+
+      return datetimes
```
Comment (connortann) [inline resqpy/time_series.py:261]: Perhaps a list comprehension is simpler here?

```suggestion
      return [dt.datetime.fromisoformat(t.rstrip('z')) for t in self.timestamps]
```

File: resqpy/time_series.py [hunk 2]
```diff
@@ -364,6 +376,31 @@ def time_series_from_list(timestamp_list, parent_model = None):
...
+   for timeseries_uuid in timeseries_uuid_iter:
+       timeseriesroot=model.root(uuid=timeseries_uuid)
+       assert(rqet.node_type(timeseriesroot) == 'obj_TimeSeries')
+       
+       singlets=TimeSeries(model, time_series_root=timeseriesroot)
+       timeserieslist.append(singlets)
+       #alltimestamps.update( set(singlets.timestamps) )
+       alltimestamps.update( set(singlets.datetimes()) )
+   
+   if reverse is None:
```
Comment (connortann) [inline resqpy/time_series.py:395]: Will reverse ever be None? It seems to be defined as False above

**Comment by andy-beer** [comment]
I'm happy to drop support for python 3.6 and earlier, @connortann

**Comment by connortann** [comment]
> I'm happy to drop support for python 3.6 and earlier, @connortann

OK, dropping support in #37 . @jrt54 , once that one is merged, please rebase this branch on top of master so the CI will run with the updated configuration.

File: resqpy/time_series.py (hunk 2 above)
Comment (jrt54) [reply resqpy/time_series.py:395]: No it won't. I originally tried to allow creation of a timeseries with an argument sorting or without sorting, but it seems like the returned timeseries generated from a list of a timestamps is almost always in ascending sorted order. So I removed the (reverse) argument.

**Comment by jrt54** [comment]
Alright I rebased and pushed again, I believe that should have updated the PR

**Comment by jrt54** [comment]
@connortann @andy-beer , is there any action waiting on me to go forward with this merge? All checks passed, but I see is a warning that I'm not authorized to merge.

**Pull request merged**